### PR TITLE
chore: Setup editorconfig and pre-commit

### PR DESCRIPTION
**Pull request by MaxymVlasov**
https://github.com/SpotOnInc/helmcharts/pull/89#issuecomment-2317872283

### CODE REVIEW

File: monochart/README.md [hunk 1]
```diff
@@ -290,22 +292,22 @@ Then, we have to update `hosts` section. There are 2 notations supported.
...
           - public_paths:
             - /status
             - /api/*
     ```
 
     Live apps examples:
-    
-    - [payments-pci-data-collection-service](https://github.com/SpotOnInc/payments-pci-data-collection-service/blob/master/deploy/releases/payments-pci-data-collection-service.yaml#L38-L39)
+
+    - [payments-pci-data-collection-service](https://github.com/SpotOnInc/payments-pci-data-collection-service/blob/9f8880f275bdc053f18b629357e9d61dcc86f557/deploy/releases/payments-pci-data-collection-service.yaml#L38-L39)
```
Comment (MaxymVlasov) [inline monochart/README.md:310]: Actually, exposes link to private repo

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - name: Install shfmt
+      run: |
+        curl -L "$(curl -s https://api.github.com/repos/mvdan/sh/releases/latest | grep -o -E -m 1 "https://.+?linux_amd64")" > shfmt \
+        && chmod +x shfmt && sudo mv shfmt /usr/bin/
+
+    - name: Install shellcheck
+      run: |
+        sudo apt update && sudo apt install shellcheck
+
+    - name: Install hadolint
```
Comment (xadereq) [inline .github/workflows/pre-commit.yaml:29]: Not sure if it's used (?), couldn't find reference anywhere.

**Review by xadereq — APPROVED** [review]
LGTM

File: .github/workflows/pre-commit.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      run: |
+        curl -L "$(curl -s https://api.github.com/repos/hadolint/hadolint/releases/latest | grep -o -E -m 1 "https://.+?/hadolint-Linux-x86_64")" > hadolint \
+        && chmod +x hadolint && sudo mv hadolint /usr/bin/
```
Comment (MaxymVlasov) [inline .github/workflows/pre-commit.yaml:32]:
```suggestion
```

File: .github/workflows/pre-commit.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        token: ${{ secrets.GITHUB_TOKEN }}
+        extra_args: --color=always --show-diff-on-failure --files ${{ steps.file_changes.outputs.files }}
+    # Run only skipped checks
+    - name: Execute pre-commit check that have no auto-fixes
+      if: always()
+      uses: pre-commit/action@9b88afc9cd57fd75b655d5c71bd38146d07135fe # v2.0.3
+      env:
+        SKIP: check-added-large-files,check-merge-conflict,check-vcs-permalinks,forbid-new-submodules,no-commit-to-branch,end-of-file-fixer,trailing-whitespace,check-yaml,check-merge-conflict,check-executables-have-shebangs,check-case-conflict,mixed-line-ending,detect-aws-credentials,detect-private-key,shfmt,shellcheck
+      with:
+        extra_args: --color=always --show-diff-on-failure --files ${{ steps.file_changes.outputs.files }}
```
Comment (MaxymVlasov) [inline .github/workflows/pre-commit.yaml:56]:
```suggestion
```

File: .github/workflows/pre-commit.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Install shellcheck
+      run: |
+        sudo apt update && sudo apt install shellcheck
+
+    # Need to success pre-commit fix push
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
+    # Skip terraform_tflint which interferes to commit pre-commit auto-fixes
```
Comment (MaxymVlasov) [inline .github/workflows/pre-commit.yaml:34]:
```suggestion

```

File: .github/workflows/pre-commit.yaml (hunk 2 above)
Comment (MaxymVlasov) [reply .github/workflows/pre-commit.yaml]: Nope, removed

**Review by dalefrancum — APPROVED** [review]

**Pull request merged**